src/common/Input.js

```diff
@@ -1,39 +1,74 @@
 /* @flow */
 import React, { Component } from 'react';
-import { TextInput } from 'react-native';
+import { View, TextInput } from 'react-native';
 import { FormattedMessage } from 'react-intl';
 import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';
 
-import styles, { HALF_COLOR } from '../styles';
+import styles, { HALF_COLOR, BRAND_COLOR } from '../styles';
+import { IconCross } from '../common/Icons';
+
+type Props = {
+  style: StyleObj,
+  restProps?: any[],
+  placeholder: string,
+  clearButton: boolean,
+  textInputRef?: (component: TextInput) => void
+};
 
 export default class Input extends Component {
-  props: {
-    style: StyleObj,
-    restProps?: any[],
-    placeholder: string,
-    textInputRef?: (component: TextInput) => void
-  };
+
+  state: {
+    input: string,
+  }
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { input: '' };
+  }
 
   static defaultProps = {
     restProps: [],
     textInputRef: () => {},
   };
 
+  handleTextChange = (text: string) => {
+    this.setState({
+      input: text,
```

Comment:
If you use 'input' only for 'can it be cleared' logic, you can convert to 'canBeCleared: boolean' in state.